backend/dataall/db/api/share_object.py

```diff
@@ -607,9 +609,41 @@ def approve_share_object(
                 resource_type=models.DatasetTable.__name__,
             )
 
+        share.rejectPurpose = ""
+        session.commit()
+
         api.Notification.notify_share_object_approval(session, username, dataset, share)
         return share
 
```

Comment:
I see that you are checking that the caller belongs to the requesters in the resolvers, but why not decorating the method with `@has_resource_perm(permissions.SUBMIT_SHARE_OBJECT)`?

Reply:
I see now why you implemented this with the "requesters" checker and the other with the permissions decorator checker! But then what I think is that the `permissions.SUBMIT_SHARE_OBJECT` is wrongly given to approvers. They are not able to see the `Submit` button in `frontend/src/views/Shares/ShareView.js` they should not be able to reach the api from the backend either. We need to modify `backend/dataall/db/permissions.py` and remove ` SUBMIT_SHARE_OBJECT,` from the list of permissions for the approver. In addition, we need to add some lines to the migration script to remove the existing permissions `SUBMIT_SHARE_OBJECT` 
```
SHARE_OBJECT_APPROVER = [
    ADD_ITEM,
    REMOVE_ITEM,
    SUBMIT_SHARE_OBJECT,
    APPROVE_SHARE_OBJECT,
    REJECT_SHARE_OBJECT,
    DELETE_SHARE_OBJECT,
    GET_SHARE_OBJECT,
    LIST_SHARED_ITEMS,
]
```
I am aware this is not the main purpose of this PR, but we could add this fix to the PR as it also includes a migration script.

Reply:
working on the revisions to the permission checker and migration script now :)